packages/driver/src/baseConn.ts

```diff
@@ -279,8 +288,15 @@ export class BaseRawConnection {
     } else {
       this.buffer.readBigInt64();
       status = this.buffer.readString();
-      this.buffer.readUUID(); // state type id
-      this.buffer.readLenPrefixedBuffer(); // state
+
+      const stateTypeId = this.buffer.readUUID();
+      const stateData = this.buffer.readLenPrefixedBuffer();
+
+      if (this.cliMode && stateTypeId === this.stateCodec.tid) {
+        this.lastStateUpdate = this.stateCodec.decode(
+          new ReadBuffer(stateData)
+        );
+      }
```

Comment:
Some protocol context: Normally for clients the session state (ie. globals, aliases and session config (like `apply_access_policies`)) is managed entirely client side via the `client.with*` api's, then encoded and sent with every query (since the protocol is stateless except for transactions). However for the cli, there's obviously no programmatic api to set this state, so instead you use the `set global/alias/module ...` and `configure session ...` commands in edgeql, and when you run one of these commands the server returns an updated session state object, which the cli stores and sends back with following queries.
And the context for this PR: Currently the UI repl follows the client behaviour, where session state is set entirely in the UI. But we want to allow the edgeql commands to work, so this PR allows the private api's the ui uses to enable this hybrid client/cli behaviour by:
- Allowing the `rawParse`/`rawExecute` methods to set custom capabilities flags, so the server won't reject the edgeql commands
- Actually decoding the updated state the server returns. I'm just storing this updated state in a `lastStateUpdate` property on the connection (similar to `lastStatus`), mainly so I don't add any complexity by having to thread it through as a return value along with the actual result from the execute method.
- Adding a `cliMode` flag so the above decoding is only enabled for the ui. And also it disables an optimisation where we don't send the session state if nothing's been set, since we need the server to send an updated codec to decode the session state, as we actually need to read it now.

Reply:
Great, thanks for the additional context! 🙏